app/services/data_centers/change_validator_data_center.rb

```diff
@@ -0,0 +1,217 @@
+
+      assign_validator_to_new_data_center
+
+      # This step needs to be removed after removing data_center_key from validator_score_v1
+      update_data_center_host_in_score 
+    end
+
+    private
+
+    def find_validator
+      @validator = Validator.find_by(id: @validator_id)
+
+      if @validator
+        log_message("Validator #{@validator.name} (##{@validator.id}) found.")
+      else
+        log_message(
+          "Validator with id: #{@validator_id} has not been found, script is finished.",
+          type: :error
+        )
+      end
+    end
+
+    def find_validator_ip
+      @validator_ip = @validator.validator_ip_active
+
+      if @validator_ip
+        message = <<-EOS
+          Validator Ip for validator #{@validator.name} 
+          (##{@validator.id}) has been found, 
+          ip address: #{@validator_ip.address}
+        EOS
+
+        log_message(message)        
+      else
+        message = <<-EOS
+          Validator Ip (#{@validator_ip.id}) for validator #{@validator.name} 
+          (##{@validator.id}) has not been found, script is finished."
+        EOS
+
+        log_message(message, type: :error)
+
+        return false
+      end
+    end
+
+    def find_data_center_host
+      @data_center_host = @validator_ip.data_center_host
+    end
+
+    def max_mind_results_for_ip
+      client = MaxMind::GeoIP2::Client.new(
+        account_id: Rails.application.credentials.max_mind[:account_id],
+        license_key: Rails.application.credentials.max_mind[:license_key]
+      )
+
+      @maxmind_results = client.insights(@validator_ip.address)
+    end
+
+    def data_center_from_maxmind_data
+      @data_center = DataCenter.find_or_initialize_by(
+        continent_code: @maxmind_results.continent.code,
+        continent_geoname_id: @maxmind_results.continent.geoname_id,
+        continent_name: @maxmind_results.continent.name,
+        country_iso_code: @maxmind_results.country.iso_code,
+        country_geoname_id: @maxmind_results.country.geoname_id,
+        country_name: @maxmind_results.country.name,
+        registered_country_iso_code: @maxmind_results.registered_country.iso_code,
+        registered_country_geoname_id: @maxmind_results.registered_country.geoname_id,
+        registered_country_name: @maxmind_results.registered_country.name,
+        traits_anonymous: @maxmind_results.traits.anonymous?,
+        traits_hosting_provider: @maxmind_results.traits.hosting_provider?,
+        traits_user_type: @maxmind_results.traits.user_type,
+        traits_autonomous_system_number: @maxmind_results.traits.autonomous_system_number,
+        traits_autonomous_system_organization: @maxmind_results.traits.autonomous_system_organization,
+        city_name: @maxmind_results.city.name,
+        location_metro_code: @maxmind_results.location.metro_code,
+        location_time_zone: @maxmind_results.location.time_zone,
+      )
+
+      @data_center.city_confidence = @maxmind_results.city.confidence
```

Comment:
What do you think about using tap?
With the tap we can also use unless condition
```suggestion
      @data_center.tap do |d|
         d.city_confidence = @maxmind_results.city.confidence
         d.city_geoname_id = @maxmind_results.city.geoname_id
       ...
```